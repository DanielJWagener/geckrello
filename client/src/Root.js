// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
