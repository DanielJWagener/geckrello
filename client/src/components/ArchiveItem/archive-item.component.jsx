import React from "react";
import { connect } from "react-redux";
import { restoreList, restoreCard } from "../../actions";

const ArchiveItem = props => {
  const restore = () => {
    if (props.itemType === "list") {
      props.restoreList(props.itemId);
    } else if (props.itemType === "card") {
      props.restoreCard(props.itemId);
    }
  };

  return (
    <div className="archive__item">
      <div className="archive__item--title">{props.itemTitle}</div>
      <button className="archive__item--restore" onClick={restore}>
        Restore
      </button>
    </div>
  );
};

export default connect(null, { restoreList, restoreCard })(ArchiveItem);
