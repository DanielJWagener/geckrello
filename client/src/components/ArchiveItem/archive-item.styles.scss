&__item {
  display: flex;
  justify-content: space-around;
  align-items: flex-center;
  margin-bottom: 2rem;

  &--title {
    font-size: 2rem;
    padding: 1em 4rem 1rem 4rem;
    text-align: center;
    background-color: #ddd;
    border-radius: 5px;
    width: 50%;
  }

  &--restore {
    font-size: 2rem;
    padding: 0.5rem 1rem 0.5rem 1rem;
    text-align: center;
    background-color: blue;
    color: white;
    border-radius: 5px;
    cursor: pointer;
    width: 25%;
  }
}
