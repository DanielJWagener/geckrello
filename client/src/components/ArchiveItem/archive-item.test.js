import { shallow } from "enzyme";
import React from "react";
import ArchiveItem from "./archive-item.component";
import Root from "../../Root";

it("renders ArchiveItem component", () => {
  expect(
    shallow(
      <Root>
        <ArchiveItem />
      </Root>
    )
  ).toMatchSnapshot();
});
