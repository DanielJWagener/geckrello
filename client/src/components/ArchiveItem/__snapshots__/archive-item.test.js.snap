// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ArchiveItem component 1`] = `ShallowWrapper {}`;
